codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/HuffmanDecoder.java

```diff
@@ -213,4 +139,113 @@ private static void insert(Node root, int symbol, int code, byte length) {
             current.children[i] = terminal;
         }
     }
+
+    private static final class DecoderProcessor implements ByteProcessor {
+        private final int initialCapacity;
+        private byte[] bytes;
+        private int index;
+        private Node node;
+        private int current;
+        private int currentBits;
+        private int symbolBits;
+
+        DecoderProcessor(int initialCapacity) {
+            this.initialCapacity = initialCapacity;
+        }
+
+        void reset() {
+            node = ROOT;
+            current = 0;
+            currentBits = 0;
+            symbolBits = 0;
+            bytes = new byte[initialCapacity];
+            index = 0;
+        }
+
+        /*
+         * The idea here is to consume whole bytes at a time rather than individual bits. node
+         * represents the Huffman tree, with all bit patterns denormalized as 256 children. Each
+         * child represents the last 8 bits of the huffman code. The parents of each child each
+         * represent the successive 8 bit chunks that lead up to the last most part. 8 bit bytes
+         * from buf are used to traverse these tree until a terminal node is found.
+         *
+         * current is a bit buffer. The low order bits represent how much of the huffman code has
+         * not been used to traverse the tree. Thus, the high order bits are just garbage.
+         * currentBits represents how many of the low order bits of current are actually valid.
+         * currentBits will vary between 0 and 15.
+         *
+         * symbolBits is the number of bits of the the symbol being decoded, *including* all those
+         * of the parent nodes. symbolBits tells how far down the tree we are. For example, when
+         * decoding the invalid sequence {0xff, 0xff}, currentBits will be 0, but symbolBits will be
+         * 16. This is used to know if buf ended early (before consuming a whole symbol) or if
+         * there is too much padding.
+         */
+        @Override
+        public boolean process(byte value) throws IOException {
+            current = (current << 8) | (value & 0xFF);
+            currentBits += 8;
+            symbolBits += 8;
+            // While there are unconsumed bits in current, keep consuming symbols.
+            while (currentBits >= 8) {
```

Comment:
we just increased `currentBits` by 8 ... can this be a `do { } while(..)` loop?

/cc @carl-mastrangelo